dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -357,10 +357,20 @@ public GitHubJsonStore getGitHubJsonStore(String store) {
     }
 
     public void createPullReq(GHRepository origRepo,
-                              String branch, GHRepository forkRepo,
-                              PullRequestInfo pullRequestInfo) throws InterruptedException, IOException {
+                              String branch,
+                              GHRepository forkRepo,
+                              PullRequestInfo pullRequestInfo,
+                              RateLimiter rateLimiter) throws InterruptedException, IOException {
         // TODO: This may loop forever in the event of constant -1 pullRequestExitCodes...
         while (true) {
+            if(rateLimiter != null) {
```

Comment:
Suggestion.  Instead of checking for `null` instead assign a NullRateLimiter whose implementation of consume just returns immediately.
```
final RateLimiter localRateLimiter = Optional.ofNullable(rateLimiter).orElse(new NullRateLimiter())
log.info("Trying to consume a token before creating pull request..");
// Consume a token from the token bucket.
// If a token is not available this method will block until
// the refill adds one to the bucket.
rateLimiter.consume();
log.info("Token consumed, proceeding with PR creation..");
``` 
Then we don't need this branching logic.

Reply:
Two reasons why Optional wasn't used here. 
approach 1:  Optional with object value or null, in this case, we'll have to pass around the optional as a method argument because of how it's being used in the code, if we choose not to do that we will end up having similar logic. 
approach 2:  Use a custom/Null rate limiter that has infinite tokens to consume. I think that's an additional code invocation for something that we don't even need if it's disabled.
I would preferred to use it if we were not passing the return value as arguments to other methods.

Reply:
open to thoughts

Reply:
We can add a TODO: to clean this up later - I don't think it is too much of a big deal as is - keeping the current change smaller is kind of nice too